integrations/GA/browser.js

```diff
@@ -29,88 +56,638 @@ class GA {
+              category: item.properties.category || props.category,
+              list: props.list_id || props.category || "products",
+              brand: item.properties.band,
+              variant: item.properties.variant,
+              price: item.properties.price,
+              position: getProductPosition(item, products),
+            };
+            impressionObj = extend(
+              impressionObj,
+              metrics(
+                item.properties,
+                dimensionsArray,
+                metricsArray,
+                contentGroupingsArray
+              )
+            );
+
+            for (var prop in impressionObj) {
+              if (impressionObj[prop] === undefined) delete impressionObj[prop];
+            }
+            ga("ec:addImpression", impressionObj);
+          });
+          pushEnhancedEcommerce(rudderElement, this.inputs);
+          break;
+        case "Product List Filtered":
+          var props = rudderElement.message.properties;
+          var products = props.products;
+          props.filters = props.filters || [];
+          props.sorters = props.sorters || [];
+          var filters = props.filters
+            .map(function (obj) {
+              return obj.type + ":" + obj.value;
+            })
+            .join();
+          var sorts = props.sorters
+            .map(function (obj) {
+              return obj.type + ":" + obj.value;
+            })
+            .join();
+
+          this.enhancedEcommerceLoaded = loadEnhancedEcommerce(
+            rudderElement,
+            this.enhancedEcommerceLoaded
+          );
+
+          each(products, function (product) {
+            var item = { properties: product };
+
+            if (
+              !(item.properties.product_id || item.properties.sku) &&
+              !item.properties.name
+            ) {
+              return;
+            }
+
+            var impressionObj = {
+              id: item.properties.product_id || item.sku,
+              name: item.name,
+              category: item.category || props.category,
+              list: props.list_id || props.category || "search results",
+              brand: props.brand,
+              variant: filters + "::" + sorts,
+              price: item.price,
+              position: getProductPosition(item, products),
+            };
+
+            impressionObj = extend(
+              impressionObj,
+              metrics(
+                item.properties,
+                dimensionsArray,
+                metricsArray,
+                contentGroupingsArray
+              )
+            );
+
+            for (var prop in impressionObj) {
+              if (impressionObj[prop] === undefined) delete impressionObj[prop];
+            }
+            ga("ec:addImpression", impressionObj);
+          });
+          pushEnhancedEcommerce(rudderElement, this.inputs);
+          break;
+        default:
+          var contextOpts; //need to implement
```

Comment:
what is the use of contextOpts and interfaceOpts?
is it required?

Reply:
removed it